src/pkg/services/m365/api/lists_test.go

```diff
@@ -267,3 +651,180 @@ func (suite *ListsAPIIntgSuite) TestLists_GetListByID() {
 		})
 	}
 }
+
+func (suite *ListsAPIIntgSuite) TestLists_PostList() {
+	t := suite.T()
+
+	ctx, flush := tester.NewContext(t)
+	defer flush()
+
+	var (
+		acl      = suite.its.ac.Lists()
+		siteID   = suite.its.site.id
+		listName = testdata.DefaultRestoreConfig("list_api_post_list").Location
+	)
+
+	writer := kjson.NewJsonSerializationWriter()
+	defer writer.Close()
+
+	oldListID := "old-list"
+	textColumnDefID := "list-col1"
+	textColumnDefName := "itemName"
+	template := "genericList"
+
+	listInfo := models.NewListInfo()
+	listInfo.SetTemplate(&template)
+
+	textColumn := models.NewTextColumn()
+
+	txtColumnDef := models.NewColumnDefinition()
+	txtColumnDef.SetId(&textColumnDefID)
+	txtColumnDef.SetName(&textColumnDefName)
+	txtColumnDef.SetText(textColumn)
+
+	list := models.NewList()
+	list.SetId(&oldListID)
+	list.SetColumns([]models.ColumnDefinitionable{txtColumnDef})
+	list.SetList(listInfo)
+
+	err := writer.WriteObjectValue("", list)
+	require.NoError(t, err)
+
+	oldListByteArray, err := writer.GetSerializedContent()
+	require.NoError(t, err)
+
+	newList, err := acl.PostList(ctx, siteID, listName, oldListByteArray)
+	require.NoError(t, err, clues.ToCore(err))
+	assert.Equal(t, listName, ptr.Val(newList.GetDisplayName()))
+
+	// clean up
+	defer func(sID string, lst models.Listable) {
+		err = acl.DeleteList(ctx, sID, ptr.Val(lst.GetId()))
+		require.NoError(t, err)
+	}(siteID, newList)
+
+	_, err = acl.PostList(ctx, siteID, listName, oldListByteArray)
+	require.Error(t, err)
+}
+
+func (suite *ListsAPIIntgSuite) TestLists_PostListItem() {
+	t := suite.T()
+
+	ctx, flush := tester.NewContext(t)
+	defer flush()
+
+	var (
+		acl      = suite.its.ac.Lists()
+		siteID   = suite.its.site.id
+		listName = testdata.DefaultRestoreConfig("list_api_post_list").Location
+	)
+
+	writer := kjson.NewJsonSerializationWriter()
+	defer writer.Close()
+
+	oldListID := "old-list"
+	listItemID := "list-item1"
+	textColumnDefID := "list-col1"
+	textColumnDefName := "itemName"
+
+	textColumn := models.NewTextColumn()
+
+	txtColumnDef := models.NewColumnDefinition()
```

Comment:
This is a lot of chaff code.  Is there any way to centralize it, or parts of it, for ease of re-use across multipl tests?